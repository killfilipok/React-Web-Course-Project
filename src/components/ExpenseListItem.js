import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'

export default ({ description, amount, createdAt, id }) => (
    <div>
        <Link to={`/edit/${id}`}>Expense List Item: </Link>
        <p>
             {description} 
            - 
            {numeral(amount /100).format('$0,0.00')} 
            created at: {moment(createdAt).format('MMMM Do, YYYY')}
        </p>
    </div>
)

