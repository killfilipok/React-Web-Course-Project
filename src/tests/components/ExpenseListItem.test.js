import ExpenseListItem from '../../components/ExpenseListItem'
import {shallow} from 'enzyme'
import React from 'react'
import expenses from '../fixtures/expenses'

test('should render expense list item', () => {
    const wrapper = shallow(<ExpenseListItem key={expenses[0].id} {...expenses[0]}/>)

    expect(wrapper).toMatchSnapshot()
})