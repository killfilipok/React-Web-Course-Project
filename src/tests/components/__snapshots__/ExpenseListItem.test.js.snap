// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    Expense List Item: 
  </Link>
  <p>
    Gum 
     amount: $2.00 created at: January 1st, 1970
  </p>
</div>
`;
